python/dask_cudf/dask_cudf/tests/test_groupby.py

```diff
@@ -246,6 +246,7 @@ def test_groupby_string_index_name(myindex):
 @pytest.mark.parametrize(
     "agg_func",
     [
+        lambda gb: gb.agg({"c": ["count"]}, split_out=2),
```

Comment:
Can we also add tests to `test_reset_index_multiindex` since these issues were popping up after calling `reset_index`.

The issue also presented itself if grouping by multiple columns and calling an aggregate in list form `.agg{'col_name':[aggs]}`. 

The pr seems to fix the single column groupby case but throws an error with multicolumn groupby.
```
df = dask_cudf.from_cudf(cudf.DataFrame({'a':[1,1,2,3,4],'b':[5,2,1,2,5],'c':[1,2,2,3,5]}),npartitions=1)
df_groups = df.groupby(['a','c']).agg({'b': ["count"]}).reset_index().compute()
```

```
ValueError: The columns in the computed data do not match the columns in the provided metadata
  Extra:   [('a', ''), ('b', 'count'), ('c', '')]
  Missing: [0, 'a', 'c']
```

Reply:
Added this test